### PR TITLE
Add support for additional Kafka listener properties

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves: https://github.com/spring-projects/spring-boot/issues/11257

### CODE REVIEW

**Comment by snicoll** [comment]
@garyrussell thanks for the PR, I've polished it a bit (see https://github.com/spring-projects/spring-boot/commit/c4db22007c2051284f1f40eacb4ab0285dc2ee1e). In particular:

* We use now `Duration` for every time-based property. If the default unit is not ms we have `DefaultDurationUnit` to precise what the default is
* We've polished the configuration key descriptions recently and I've adapted those keys to it. It would be great if you had a quick look to the polish commit or the appendix to review that

Also, we are now running against Kafka `2.1.1.BUILD-SNAPSHOT`. Could you please set a release date for it so that we can confirm we're good for Spring Boot `2.0.0.RC1` (see #11524).

Thanks!

**Comment by garyrussell** [comment]
Thanks; done; I scheduled it for next Friday.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/ConcurrentKafkaListenerContainerFactoryConfigurer.java [hunk 1]
```diff
@@ -101,6 +101,11 @@ private void configureContainer(ContainerProperties container) {
 				.to(container::setAckTime);
 		map.from(properties::getPollTimeout).whenNonNull().as(Duration::toMillis)
 				.to(container::setPollTimeout);
+		map.from(properties::getClientId).whenNonNull().to(container::setClientId);
```
Comment (idolice) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/ConcurrentKafkaListenerContainerFactoryConfigurer.java]: Hi, I could not find the method setClientId for my ContainerProperties, I think it caused some issues

Comment (snicoll) [reply spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/ConcurrentKafkaListenerContainerFactoryConfigurer.java]: If you do, you are overriding the managed version for Spring Kafka in your application. Please remove the override or upgrade to the snapshot for the time being. Spring Kafka `2.1.1.RELEASE` will have this property.

**Pull request closed without merging**